### PR TITLE
Add Half library to third_party

### PR DESCRIPTION
**Pull request by ThomasRaoux**
Add half library to be able to use it for conversion between half and float in iree tools.

### CODE REVIEW

**Review by benvanik — CHANGES_REQUESTED** [review]
I think I missed the part of the convo earlier - this has no git repo at all anywhere? If it does this should be a submodule. Otherwise, if it's like one file out of a 10000k file project then it's ok to clone in here, but we need to include updating instructions/shell script that fetches a new version (like `UPDATING.md` or `update.sh`, etc)

**Comment by benvanik** [comment]
(and to clarify, that updating script doesn't need to be heavyweight - it could literally be `curl https://raw.githubusercontent.com/whoever/project/master/half.hpp > half.hpp`)

**Comment by ThomasRaoux** [comment]
> I think I missed the part of the convo earlier - this has no git repo at all anywhere? If it does this should be a submodule. Otherwise, if it's like one file out of a 10000k file project then it's ok to clone in here, but we need to include updating instructions/shell script that fetches a new version (like `UPDATING.md` or `update.sh`, etc)
> 
> (these are the same rules we have internally in //third_party - knowing where a file came from and how to update it is very very useful :)

It comes from sourceforge and is in an svn repo so I cannot make it a submodule. I added a file UPDATING.md with information on how to update this project.

**Review by benvanik — APPROVED** [review]
Thanks!

(sourceforge... now there's a name I haven't heard in awhile :)

**Review by ScottTodd — APPROVED** [review]
format check is failing despite https://github.com/google/iree/blob/347d5c49321def2bcfc8252f3057afead9341b68/third_party/.clang-format#L15-L16 @GMNGeoffrey ?

File: third_party/half/UPDATING.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+This project is from: https://sourceforge.net/p/half/code/HEAD/tree/
+
+This can be updated by running:
+```shell
+curl http://svn.code.sf.net/p/half/code/trunk/include/half.hpp > half.hpp
```
Comment (GMNGeoffrey) [inline third_party/half/UPDATING.md:5]: We should be updating the license and README as well, I think

**Review by GMNGeoffrey — COMMENTED** [review]
> format check is failing despite
> 
> https://github.com/google/iree/blob/347d5c49321def2bcfc8252f3057afead9341b68/third_party/.clang-format#L15-L16
> 
> @GMNGeoffrey ?

:angry: I'll try to debug. Best guess is that git-clang-format targets individual files and so ignores the part of the style file that says to ignore third_party.

File: third_party/half/UPDATING.md (hunk 1 above)
Comment (ThomasRaoux) [reply third_party/half/UPDATING.md:5]: Done.

**Pull request merged**